File: src/bronnen/Afval.cpp
#include "Afval.h"

Afval::Afval() : Bron::Bron()
{
	// Dag van de week
	time_t t = time(NULL);
	tm* timePtr = localtime(&t);

	dagNu = timePtr->tm_wday;

	// Afval dagen
	afvalDag.insert(pair<int, string>(1, "plastic")); 
	afvalDag.insert(pair<int, string>(4, "papier")); 

	// Zoek afval voor vandaag op
	itVandaag = afvalDag.find(dagNu);
	if (itVandaag != afvalDag.end()) 
	{
		text = "Vandaag moet het " + itVandaag->second + " bij de weg.";
		hasText = true;
	}
}

bool Afval::HasText()
{
	return hasText;
}

string Afval::GetText()
{
	return text;
}

File: modules/Afval.py
import datetime, json
from lib import Functions as f
from modules import _Module

class Afval(_Module):

	def Run(self):
		now = f.Now(self.config_full)

		# Na 11u is het toch al te laat
		if now.hour < 11:
			# Dag van vandaag
			weekday = now.weekday()

			# Staat vandaag in de config file?
			if str(weekday) in self.config:
				self.hasText = True
				self.text = "Vandaag moet het " + self.config[str(weekday)] + " bij de weg"
